Sources/Corvus/Endpoints/SecureCRUD.swift

```diff
@@ -0,0 +1,94 @@
+import Vapor
+import Fluent
+
+/// A class that contains Create, Read, Update and Delete functionality for a
+/// generic type `T` conforming to `CorvusModel` grouped under a given path and
+/// allows securing those resources with the `.auth()` modifier.
+public final class SecureCRUD<
```

Comment:
I would suggest to make `SecureCRUD ` and internal type and only expose it using the auth modifier and package the return of that modifier in a protocol that CRUD and SecureCRUD conform to (maybe using an opaque return type?) or make SecureCRUD just a subclass of CRUD. This would lower the API surface a bit and make it common to the other usages of the auth modifier and brings the modifier to all CRUD operations + CRUD itself.

Reply:
Made `SecureCRUD` a subclass of `CRUD`.